localstack/services/awslambda/lambda_api.py

```diff
@@ -1013,31 +988,31 @@ def not_found_error(ref=None, msg=None):
 
 
 def delete_lambda_function(function_name: str) -> Dict[None, None]:
-    region = LambdaRegion.get_for_arn(function_name)
+    store = get_awslambda_store_for_arn(function_name)
     arn = func_arn(function_name)
     # Stop/remove any containers that this arn uses.
     LAMBDA_EXECUTOR.cleanup(arn)
 
     try:
-        region.lambdas.pop(arn)
+        store.lambdas.pop(arn)
     except KeyError:
         raise ResourceNotFoundException(
             f"Unable to delete non-existing Lambda function {func_arn(function_name)}"
         )
 
     i = 0
-    while i < len(region.event_source_mappings):
-        mapping = region.event_source_mappings[i]
+    while i < len(store.event_source_mappings):
+        mapping = store.event_source_mappings[i]
         if mapping["FunctionArn"] == arn:
-            del region.event_source_mappings[i]
+            del store.event_source_mappings[i]
             i -= 1
         i += 1
     return {}
 
 
 def get_lambda_url_config(api_id: str, region: str = None):
-    lambda_backend = LambdaRegion.get(region)
-    url_configs = lambda_backend.url_configs.values()
+    store = get_awslambda_store()
```

Comment:
We are not using the region parameter here at all anymore. Even though it might be extractable from the contextvars (or thread locals I think), would be better to explicitly pass them here, I think.

Reply:
Good catch, thanks! fixed in ee824ded3d6d5260b63839df59baadcdfdd1c645